PANA-EXP - Adding docs for the experimentation feature (#32587)
DataDog/documentation: content/en/product_analytics/experimentation/defining_metrics.md

---
title: Defining Metrics
description: Define the metrics you want to measure during your experimentation.
further_reading:
- link: "https://www.datadoghq.com/blog/datadog-product-analytics/"
  tag: "Blog"
  text: "Make data-driven design decisions with Product Analytics"
- link: "/product_analytics/experimentation/reading_results.md"
  tag: "Documentation"
  text: "Reading Experiment Results"
---

## Overview

Define the metrics you want to measure during your experimentation. Metrics can be built using Real User Monitoring (RUM) and Product Analytics data.

> In order to create a metric, you must have Datadog’s client SDK installed in your application and be actively capturing data.

Reviewer: This also feels like something impt that could go in a pre-reqs section or a callout/note ?

## Create your first metric

To create a metric for your experiments: 

1. Navigate to the [Metrics page][1] within Datadog Product Analytics.
2. Click **“Create Metric”** at the top right corner. 
3. Click **“Select an Event”** to see a list of all of the actions and views collected from the RUM SDK. 

## Specify metric aggregation

### Default metric aggregation 

> After you’ve selected your event of interest, you can specify an aggregation method. Metrics default to a **count of unique** subjects (often users) that had at least one event. But, you can also choose to _count the total number_ of events, or _sum a property_ of that event:

Reviewer: ```suggestion
After you’ve selected your event of interest, you can specify an aggregation method. Metrics default to a **count of unique** subjects (often users) that had at least one event. However, you can also choose to _count the total number_ of events, or _sum a property_ of that event:
```

> {{< img src="dd-logo.png" alt="A view of the User profiles page." style="width:10%;" >}}

Reviewer: ```suggestion
{{< img src="dd-logo.png" alt="A view of the User profiles page." style="width:10%;" >}}
```
same comment for this image

Reviewer 2: This UI should be ready for a screenshot now (https://demo.datadoghq.com/product-analytics/experimentation-metrics/create)

<img width="738" height="284" alt="image" src="https://github.com/user-attachments/assets/b75c94df-9a8f-4bd9-a622-772e93648c19" />

Similarly, a **Sum of** metric is computed as: 

<!-- {{< img src="/product_analytics/experiment/pana_exp_equation2.png" alt="A view of the User profiles page." style="width:60%;border:none" >}} -->

```

            Sum of property over users enrolled in this variant
Sum of =  --------------------------------------------------------           
                Number of users enrolled into this variant
```

### Custom normalized metrics

You can also choose to normalize metrics by a different denominator. To do this, click **Create ratio** `WHERE?`. This allows you to normalize your metric by another event, counting either the number of subjects with that event, the total number of events, or the sum of an event property. 

For example, an e-commerce company that wants to measure the _Average Order Value_ can create a ratio metric with the sum of purchase revenue as the numerator, and the count of purchase events as the denominator. 

Datadog’s statistical engine accounts for correlations between the numerator and denominator using the [delta method][2]. 

## Add filters

> You can also add filters (`to your metrics??`), similar to other Product Analytics dashboards. For instance, you might want to filter page views based on referring URL or UTM parameters. Similarly, you might want to filter actions to a specific page or value of a custom attribute.

Reviewer: ```suggestion
You can also add filters to your metrics, similar to other Product Analytics dashboards. For instance, you might want to filter page views based on referring URL or UTM parameters. Similarly, you might want to filter actions to a specific page or value of a custom attribute.
```

{{< img src="dd-logo.png" alt="A view of the User profiles page." style="width:10%;" >}}

> As you add filters, you can check metric values in real time using the chart on the right (`WHERE??`).

Reviewer: ```suggestion
As you add filters, you can check metric values in real time using the chart on the right (`WHERE??`).
```
Just adding a note that you are addressing this "Where" comment ? :)

## Advanced options
Datadog supports several advanced options specific to experimentation:

`Timeframe filters`
: - By default, Datadog will include all events between a user's first exposure and the end of the experiment. If you want to measure a time-boxed value such as “sessions within 7 days”, you can add a timeframe filter.
  - If selected, the metric will only include events from the specified time window, starting at the moment the user was first enrolled.

`Desired metric change`
: - Datadog highlights statistically significant results. 
  - Use this setting to specify whether an increase or decrease in this metric is desired.

`Guardrail cutoffs`
: - Use this to specify a `do no harm` threshold. 

>   - Metrics will be yellow(`REPRESENTING WHAT?`) until the confidence interval is small enough to rule out a downside risk larger than this threshold.

Reviewer: ```suggestion
  - Metrics will be yellow(`REPRESENTING WHAT?`) until the confidence interval is small enough to rule out a downside risk larger than this threshold.
```
Same with this one

                      Number of users enrolled into this variant
```

Similarly, a **Sum of** metric is computed as: 

```

                  Sum of property over users enrolled in this variant
              --------------------------------------------------------           
                      Number of users enrolled into this variant
```

### Custom metric normalization

You can also choose to normalize metrics by a different denominator. To do this, click on a metric
then on **Create ratio**. This allows you to normalize your metric by another event, counting either the number of subjects with that event, the total number of events, or the sum of an event property. 

{{< img src="/product_analytics/experiment/exp_create_ratio.png" alt="Button used to create a ratio from the metric." style="width:90%;" >}}

For example, an e-commerce company that wants to measure the _Average Order Value_ can create a ratio metric with the sum of purchase revenue as the numerator, and the count of purchase events as the denominator. 

Datadog’s statistical engine accounts for correlations between the numerator and denominator using the [delta method][2]. 

## Add filters

> You can also add filters to your metrics, similar to other Product Analytics dashboards. For instance, you might want to filter page views based on referring URL or UTM parameters. Similarly, you might want to filter actions to a specific page or value of a custom attribute. As you add filters, you can check metric values in real time using the chart on the right.

Reviewer: ```suggestion
You can also add filters to your metrics, similar to other [Product Analytics dashboards][]. For instance, you might want to filter page views based on referring URL or UTM parameters. Similarly, you might want to filter actions to a specific page or value of a custom attribute. As you add filters, you can check metric values in real time using the chart on the right.
```
Not a blocker but it could be nice to link to Product analytics dashboard docs here ?

---
title: Defining Metrics
description: Define the metrics you want to measure during your experimentation.
further_reading:
- link: "https://www.datadoghq.com/blog/datadog-product-analytics/"
  tag: "Blog"
  text: "Make data-driven design decisions with Product Analytics"
- link: "/product_analytics/experimentation/reading_results.md"
  tag: "Documentation"
  text: "Reading Experiment Results"
---

## Overview

> Define the metrics you want to measure during your experimentation. Metrics can be built using Real User Monitoring (RUM) and Product Analytics data.

Reviewer 2: ```suggestion
Define the metrics you want to measure during your experimentation. Metrics can be built using Product Analytics and Real User Monitoring (RUM) data.
```

<div class="alert alert-info"> In order to create a metric, you must have Datadog’s client SDK installed in your application and be actively capturing data.
</div>

## Create your first metric

To create a metric for your experiment: 

1. Navigate to the [Metrics page][1] within Datadog Product Analytics.
2. Click **+ Create Metric** at the top right corner. 

> 3. Click **Select an Event** to see a list of all of the actions and views collected from the RUM SDK.

Reviewer 2: ```suggestion
3. Click **Select an Event** to see a list of all of the actions and views collected from the Datadog SDK.
```

4. Add a metric name and optionally a description. Then, click **Save**.

> {{< img src="/product_analytics/experiment/exp_create_metric.png" alt="create an experiment and add a hypothesis for the experiment." style="width:80%;" >}}

Reviewer 2: Can we drop the red square around "Guardrail Cutoff"?

## Specify metric aggregation

### Default metric aggregation 
After you’ve selected your event of interest, you can specify an aggregation method. Metrics default to a **count of unique** subjects (often users) that had at least one event. However, you can also choose to _count the total number_ of events, or _sum a property_ of that event:

{{< img src="dd-logo.png" alt="A view of the User profiles page." style="width:10%;" >}}

> ### Default normalized metrics 

Reviewer 2: ```suggestion
### Default metric normalization 
```

All metrics are normalized by the number of enrolled subjects. For example, a **count of unique users** metric is computed as: 

```

                             Number of users with specified event
Count of unique users = -----------------------------------------------           
                          Number of users enrolled into this variant
```

<!-- {{< img src="/product_analytics/experiment/pana_exp_equation1.png" alt="A view of the User profiles page." style="width:50%;border:none" >}} -->

Similarly, a **Sum of** metric is computed as: 

<!-- {{< img src="/product_analytics/experiment/pana_exp_equation2.png" alt="A view of the User profiles page." style="width:60%;border:none" >}} -->

```

            Sum of property over users enrolled in this variant

> Sum of =  --------------------------------------------------------           

Reviewer 2: ```suggestion
--------------------------------------------------------           
```

(for consistency)

                Number of users enrolled into this variant
```

> ### Custom normalized metrics

Reviewer 2: ```suggestion
### Custom metric normalization
```

{{< img src="/product_analytics/experiment/exp_filter_by.png" alt="Filter your metric my specific properties." style="width:100%;" >}}

## Advanced options
Datadog supports several advanced options specific to experimentation:

`Timeframe filters`
: - By default, Datadog will include all events between a user's first exposure and the end of the experiment. If you want to measure a time-boxed value such as “sessions within 7 days”, you can add a timeframe filter.
  - If selected, the metric will only include events from the specified time window, starting at the moment the user was first enrolled.

`Desired metric change`
: - Datadog highlights statistically significant results. 
  - Use this setting to specify whether an increase or decrease in this metric is desired.

`Guardrail cutoffs`
: - Use this to specify a `do no harm` threshold. 

>   - Metrics will be yellow(`REPRESENTING WHAT?`) until the confidence interval is small enough to rule out a downside risk larger than this threshold.

Reviewer 2: ```suggestion
  - Metrics will be yellow until the confidence interval is small enough to rule out a downside risk larger than this threshold. This indicates that you should continue running your experiment to rule out an effect worse than the number entered here.
```
